sklearn/utils/seq_dataset.pxd

```diff
@@ -25,10 +33,6 @@ cdef class ArrayDataset(SequentialDataset):
     cdef int *feature_indices_ptr
     cdef double *sample_weight_data
 
-    cdef void next(self, double **x_data_ptr, int **x_ind_ptr,
-                   int *nnz, double *y, double *sample_weight) nogil
-
```

Comment:
I believe i added both next and random so that sgd could sample in order while sag could sample randomly. I'm away from the code right now, the tests for sgd still pass?